_extras/guide.md

```diff
@@ -27,13 +27,14 @@ is included as [`code/gen_inflammation.py`](../code/gen_inflammation.py).
 ## Overall
 
 This lesson is written as an introduction to Python,
-but its real purpose is to introduce the single most important idea in programming:
-how to solve problems by building functions,
-each of which can fit in a programmer's working memory.
-In order to teach that,
-we must teach people a little about
-the mechanics of manipulating data with lists and file I/O
-so that their functions can do things they actually care about.
+but its wider purpose is to introduce an important idea in programming:
+how to solve problems by building functions, units of code that allow
+reusability and efficient logical organization of software.
+
+So that our functions can do things we actually care about,
+we cover programming concepts such as variables, conditionals,
```

Comment:
"So that our functions can do things we actually care about,"  would've been a great intro to a paragraph describing how to write better functions when we already know how to write simple ones. I think we're one or two steps away from there. Would any of the following fit better the prose of this section?:

> To show learners how to write functions that do things they care about, we cover basic programming concepts such as variables, ...

or
> To teach how write functions that do exactly what we care about, we cover basic programming concepts such as variables, ...

Reply:
In the context of the Instructor Notes and this being just after a statement about what functions are for, I don't really think the prose options above will make for much more of a better fit for the section. However, I prefer your first option to the second (because I don't favor the use of "exactly") if others feel the wording should be changed.

Reply:
> (because I don't favor the use of "exactly")

Yeah, I think I agree with that.

Anyways, my point was that we cover programming concepts that we cover not to make our functions do things we care about but to teach/learn how to program in Python. We also learn about functions (which should be added to the list of things we learn). So, here is another possible justification:

> To teach how to write Python programs that perform calculations or carry out analysis that we need, we cover most important programming concepts such as variables, conditional statements, loops, lists, functions. We also show how to read data from files and provide input on the command line.

Reply:
Perhaps you are speaking to the point @ndilekli brought up earlier:

> the training did not have the feeling like there was a particular effort on the introduction and practice of functions.

In this case, perhaps we should take the emphasis off of functions in the previous sentence as well and say something along the lines of:

> This lesson is written as an introduction to Python,
but, its broader purpose is to introduce how to solve problems by developing programs.
By writing lines of instructions to carry out tasks, we automate our work and
enable reproducible results.
To teach how to write Python programs that perform calculations or carry out analysis that we need,
we cover most important programming concepts such as variables, conditional statements, loops,
lists, and functions.
We also show how to read data from files and provide input on the command line.

Reply:
Yes, I like this paragraph a lot more: it's easy to read and makes the right emphasis (I'm not sure about "lines of instructions" in the second sentence -- perhaps we need to clearly state that programs are lines of code/instructions -- but that's a minor detail).